core/src/main/java/org/infinispan/registry/ScopedKey.java

```diff
@@ -46,6 +47,19 @@ public int hashCode() {
       return result;
    }
 
+   @Override
+   public String toString() {
+      Object oScope = scope;
+      Object oKey = key;
+      if (oScope instanceof MarshalledValue) {
```

Comment:
Is this really necessary? AFAIK the scope and key should never be MarshalledValues.